posts/copilot-workspace/index.qmd

```diff
@@ -0,0 +1,102 @@
+
+The changes made will still require the user to manually run the documentation (using either `devtools::document()` or `roxygen2::roxygenize()`) to update the package `NAMESPACE` for the function to be exported. Our recommendation would be for AI tools to integrate with and call development tools used by developers (analogous to pre-commit hooks running tools before committing) to make sure this is taken care of all in a single generative step.
+
+## Experiment 2: Add a simple epidemic model to an R package that contains more complex models
+
+::: {.callout-note}
+Difficulty level: moderate to high
+:::
+
+{epidemics} is an Epiverse-TRACE R package that exports a range of compartmental  epidemiological models of varying complexities and applications.
+
+In the second test case, we created an issue in the {epidemics} R package requesting a basic SIR model, with the issue description requesting the model to have no age structure and to use existing functionality in the package to specify a disease intervention. Additionally, the issue requested that the model be solved with the {deSolve} R package, which is a differential equation solver in R. A basic SIR model would have been new to the package, but one of several other compartmental models, and simpler than the models already implemented.
+
+![Screenshot of epidemics package issue 238 which was used as an example application of GitHub Copilot Workspace.](epidemics_issue.png)
+
+### Outcome
+
+GitHub copilot workspace outlined the existing functionality in the package and proposed changes including added a new file called `R/model_sir.R`, which follows the `model_*.R` naming convention used in {epidemics}. As requested, copilot also proposed that the new SIR model would support the use of interventions that have already been implemented in the package.
+
+### Our thoughts
+
+The code generated was an R function that constructed a basic SIR model, and this was well documented using a {roxygen2} documentation block. However, several aspects of the code generated did not match the proposal generated by copilot workspace, and the code contained inadvisable coding practices in R (i.e. code smells). Firstly the SIR model itself. The model implemented follows the standard set of differential equations that are solved using {deSolve}, as requested in our prompt. However, the model does not have any options to input interventions, which the proposed solution by GitHub copilot workspace suggested it would. The second downside is the use of `require(deSolve)` in the function body of the generated code. This is bad practice in R as functions should namespace any functions that are being called from other packages (e.g. `packagename::func_name()`).
```

Comment:
```suggestion
The code generated was an R function that constructed a basic SIR model, and this was well documented using a {roxygen2} documentation block. However, several aspects of the code generated did not match the proposal generated by copilot workspace, and the code contained inadvisable coding practices in R (i.e. code smells). Firstly the SIR model itself. The model implemented follows the standard set of differential equations that are solved using {deSolve}, as requested in our prompt. However, the model does not have any options to input interventions, which the proposed solution by GitHub copilot workspace suggested it would. The second downside is the use of `require(deSolve)` in the function body of the generated code. This is bad practice in R package development as functions should namespace any functions that are being called from other packages (e.g. `packagename::func_name()`).
```